crates/miden-lib/asm/kernels/transaction/lib/output_notes.masm

```diff
@@ -0,0 +1,60 @@
+use.$kernel::memory
+use.$kernel::note
+
+# ERRORS
+# =================================================================================================
+
+const.ERR_OUTPUT_NOTE_INDEX_OUT_OF_BOUNDS="requested output note index should be less than the total number of created output notes"
+
+# INPUT NOTE PROCEDURES
+# =================================================================================================
+
+#! Returns the information about assets in the output note with the specified index.
+#!
+#! This information can then be used to retrieve the actual assets from the advice map.
+#!
+#! Inputs:  [note_index]
+#! Outputs: [ASSETS_HASH, num_assets]
+#!
+#! Where:
+#! - note_index is the index of the output note whose assets info should be returned.
+#! - num_assets is the number of assets in the specified note.
+#! - ASSETS_HASH is a sequential hash of the assets in the specified note.
+#!
+#! Panics if:
+#! - the note index is greater or equal to the overall number of output notes.
+export.get_assets_info
+    # assert that the provided note index is less than the total number of notes
+    dup exec.memory::get_num_output_notes 
+    # => [output_notes_num, note_index, note_index]
+    
+    lt assert.err=ERR_OUTPUT_NOTE_INDEX_OUT_OF_BOUNDS
+    # => [note_index]
+
+    # get the note data pointer based on the index of the requested note
+    exec.memory::get_output_note_ptr
+    # => [note_data_ptr]
+
+    # get the number of assets in the note
+    dup exec.memory::get_output_note_num_assets swap
+    # => [note_data_ptr, num_assets]
+
+    # get the assets commitment dirty flag and decide whether we need to recompute the commitment
+    dup exec.memory::get_output_note_dirty_flag
+    # => [dirty_flag, note_data_ptr, num_assets]
+
+    if.true
+        # we should recompute assets commitment
+        dup exec.note::compute_output_note_assets_commitment
+        # => [COMPUTED_ASSETS_HASH, note_data_ptr, num_assets]
+
+        # update the dirty flag
+        push.0 movup.5 exec.memory::set_output_note_dirty_flag
+        # => [COMPUTED_ASSETS_HASH, num_assets]
```

Comment:
I think I made a mistake here: I should move the flag update inside the `compute_output_note_assets_commitment` procedure, so that the computation of the note id will update this flag too. I'll update it tomorrow.

Reply:
Yes, I think the high-level logic for `compute_output_note_assets_commitment` could be something like this:
```
if is_commitment_current == false
   re-compute and store the commitment
   set is_commitment_current = true
end

return the stored commitment
```